lib/elixir/lib/module/checker.ex

```diff
@@ -1,16 +1,212 @@
 defmodule Module.Checker do
-  def verify(module_map, _binary) do
+  @moduledoc false
+
+  defmodule Coordinator do
+    @moduledoc false
```

Comment:
We need some docs and structure in this module. You can see other private modules, such as the lexical and locals tracker as an example, but in a nutshell.

1. Keep all client APIs at the top
2. Keep all server callbacks at the bottom
3. Document all client APIs

:)

Reply:
We should also document that pseudo code we wrote about lookup rules, what we try, in what order, and why.

Reply:
I rewrote this logic a bit to make it simpler. Will refactor when we support the mix compiler.